client/extensions/woocommerce/lib/generate-variations/index.js

```diff
@@ -0,0 +1,36 @@
+// http://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript
+const f = ( a, b ) => [].concat( ...a.map( c => b.map( d => [].concat( c, d ) ) ) );
+const cartesian = ( a, b, ...c ) => b ? cartesian( f( a, b ), ...c ) : a;
```

Comment:
I don't know if this will become a performance concern (you talked in Slack that this may be executed on every keypress when editing a product attrs). There are clearly ways to optimize.

This solution creates *a lot* of temporary arrays, which will stress the GC. It may be worth to change it to a solution without so many `.map()` and `.concat()`, like http://stackoverflow.com/a/29585704 or https://pastebin.com/YbhqZuf7